be/src/exec/pipeline/sort/local_parallel_merge_sort_source_operator.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exec/pipeline/sort/local_parallel_merge_sort_source_operator.h"
+
+#include <algorithm>
+#include <sstream>
+
+#include "exprs/expr.h"
+#include "runtime/exec_env.h"
+#include "runtime/runtime_state.h"
+
+namespace starrocks::pipeline {
+
+Status LocalParallelMergeSortSourceOperator::prepare(RuntimeState* state) {
+    RETURN_IF_ERROR(Operator::prepare(state));
+    _sort_context->ref();
+    _merger->bind_profile(_driver_sequence, _unique_metrics.get());
+    return Status::OK();
+}
+
+void LocalParallelMergeSortSourceOperator::close(RuntimeState* state) {
+    _sort_context->unref(state);
+    Operator::close(state);
+}
+
+StatusOr<ChunkPtr> LocalParallelMergeSortSourceOperator::pull_chunk(RuntimeState* state) {
+    ChunkPtr chunk = _merger->try_get_next(_driver_sequence);
+
+    if (_merger->is_finished()) {
+        _is_finished = true;
+    }
+
+    return chunk;
+}
+
+Status LocalParallelMergeSortSourceOperator::set_finished(RuntimeState* state) {
+    if (_driver_sequence == 0) {
+        return _sort_context->set_finished();
+    }
```

Comment:
why not the last?

Reply:
> why not the last?

Any of them works fine. But for simplicity, we choose the first parallelism to outout merged data.